rag-embeddings/src/main.rs

```diff
@@ -98,14 +121,29 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
 
     let client = qdrant::Qdrant::new();
 
+    let start_vector_id = match matches.get_one::<String>("start_vector_id") {
+        Some(v) => v.trim().parse().unwrap(),
+        None => 0,
+    };
     let mut id : u64 = 0;
     let mut current_section = String::new();
     let file = File::open(file_name)?;
     let reader = BufReader::new(file);
+    let mut code_mode = false;
     for line_result in reader.lines() {
         let line = line_result?;
-        if line.trim().is_empty() && (!current_section.trim().is_empty()) {
-            generate_upsert(&mut context, &current_section, &client, id, collection_name, vector_size).await;
+        if let Some(_) = line.find("```") {
```

Comment:
Perhaps this should be `trim` and then `startsWith`?

Reply:
Yes, I have fixed it. Thank you.